### PR TITLE
[FIX] web_editor: fix required html fields

### PR DESCRIPTION
**Pull request by jvnm-dev**
### Issue

    - Studio
    - Add an required html field
    - Create new record
    - Leave it empty

    It is saved

### Cause

    The isSet function of FieldHtml widget, have a "/" in the <br> tag
    and summernote empty = <p><br></p>. If you add a space, the value
    becomes <p>&nbsp;&nbsp;   <p>.

### Solution

    Remove the "/" and the spaces, check if value != <p></p> &&
    <p><br></p>

**OPW-2183687**

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/web_editor/static/tests/field_html_tests.js [hunk 1]
```diff
@@ -126,6 +132,30 @@ QUnit.module('web_editor', {}, function () {
...
+                      '</form>',
+                res_id: 1,
+            });
+
+            await testUtils.form.clickEdit(form);
+            await testUtils.nextTick();
+            await testUtils.dom.click(form.$('.o_form_button_save'));
+            await testUtils.nextTick();
+
+            assert.strictEqual(form.mode, 'edit', 'should be still in edit mode if the save fails');
```
Comment (aab-odoo) [inline addons/web_editor/static/tests/field_html_tests.js]: you can actually check that it sent a warning (by intercepting event 'call_service', 'notification' service)

File: addons/web_editor/static/tests/field_html_tests.js [hunk 2]
```diff
@@ -126,6 +132,30 @@ QUnit.module('web_editor', {}, function () {
...
+                arch: '<form>' +
+                        '<field name="header" widget="html" style="height: 100px" />' +
+                      '</form>',
+                res_id: 1,
+            });
+
+            await testUtils.form.clickEdit(form);
+            await testUtils.nextTick();
+            await testUtils.dom.click(form.$('.o_form_button_save'));
+            await testUtils.nextTick();
```
Comment (aab-odoo) [inline addons/web_editor/static/tests/field_html_tests.js]: not sure those extra next tick are necessary

Comment (jvm-odoo) [reply addons/web_editor/static/tests/field_html_tests.js]: Indeed, the second one is not necessary!

File: addons/web_editor/static/tests/field_html_tests.js [hunk 3]
```diff
@@ -126,6 +132,39 @@ QUnit.module('web_editor', {}, function () {
...
         });
 
+        QUnit.test('check if required field is set', async function (assert) {
+            assert.expect(1);
+
+            var form = await testUtils.createView({
+                View: FormView,
+                model: 'note.note',
+                data: this.data,
+                debug: true,
```
Comment (jvm-odoo) [inline addons/web_editor/static/tests/field_html_tests.js]: Oops missed that, will remove it

**Comment by jvm-odoo** [comment]
Tests should be good now

**Comment by aab-odoo** [comment]
robodoo r+

**Pull request closed without merging**